test/test_nn.py

```diff
@@ -13258,6 +13258,25 @@ def test_MarginLoss_empty(self, device, dtype):
                 y = torch.ones(10, 0, device=device).type(torch.long)
                 mod(x, y)
 
+    @dtypes(torch.float, torch.double)
+    def test_adaptive_pooling_zero_batch(self, dtype):
+        # The device parameter does not accept device objects.
+        device_list = ['cpu']
+        if TEST_CUDA:
+            device_list.append('cuda')
```

Comment:
Curious to know what's happening here, does:

```python
@onlyOnCPUAndCUDA
@dtypes(torch.float, torch.double)
def test_adaptive_pooling_zero_batch(self, dtype, device):
    # ...
```

not work? Apologies if this was answered by `The device parameter does not accept device objects` comment but I didn't completely understand this, hence the question.

Reply:
Yeah the module code seems to be outdated and only accepts devices in string form.

Reply:
Interesting, thanks for clarifying @v0dro! :)

Reply:
I don't think this is true? Test works for me as written above by @krshrimali

Reply:
Ah turns out it works on the latest master. I've changed it and pushed the code now.